File: src/vision/formExtractionEngine.ts
/**
 * Converted from Python: core/vision/form_extraction_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export let INPUT_KEYWORDS: any = new Set(["email", "password", "username", "search"]);
export function extractForms(blocks: any): any {
  var forms: any[] = [];
  var block: any;
  for (block of py.iter(py.get(blocks, "blocks", []))) {
    var text: any = String(py.get(block, "text", "")).toLowerCase();
    if (py.contains(INPUT_KEYWORDS, text)) {
      py.listAppend(forms, {"field": text, "bbox": py.get(block, "bbox")});
    }
  }
  return {"forms": forms, "bounded": true};
}

File: lib/src/vision/form_extraction_engine.dart
/// Port of core/vision/form_extraction_engine.py: extract_forms.
library;

const Set<String> inputKeywords = <String>{
  'email',
  'password',
  'username',
  'search',
};

/// Port of core/vision/form_extraction_engine.extract_forms.
Map<String, dynamic> extractForms(
  Map<String, dynamic> blocks,
) {
  final List<Map<String, dynamic>> forms = <Map<String, dynamic>>[];

  final List<dynamic> blockList = blocks.containsKey('blocks')
      ? blocks['blocks'] as List<dynamic>
      : <dynamic>[];

  for (final dynamic blockRaw in blockList) {
    final Map<String, dynamic> block = blockRaw as Map<String, dynamic>;
    final String text =
        ((block.containsKey('text') ? block['text'] : '') as String)
            .toLowerCase();

    if (inputKeywords.contains(text)) {
      forms.add(<String, dynamic>{
        'field': text,
        'bbox': block['bbox'],
      });
    }
  }

  return <String, dynamic>{
    'forms': forms,
    'bounded': true,
  };
}
